git.go

```diff
@@ -0,0 +1,244 @@
+		config.RepositoryURL != config.PRRepositoryCloneURL
+}
+
+func isPrivate() bool {
+	return strings.HasPrefix(config.PRRepositoryCloneURL, "git")
+}
+
+func autoMerge() error {
+	if err := runWithRetry(func() *command.Model {
+		if config.CloneDepth != "" {
+			return Git.Fetch("--depth=" + config.CloneDepth)
+		}
+		return Git.Fetch()
+	}); err != nil {
+		return fmt.Errorf("Fetch failed, error: %v", err)
+	}
+
+	if config.PRMergeBranch != "" {
+		if err := runWithRetry(func() *command.Model {
+			return Git.Fetch("origin", config.PRMergeBranch+":"+
+				strings.TrimSuffix(config.PRMergeBranch, "/merge"))
+		}); err != nil {
+			return fmt.Errorf("fetch Pull Request branch failed (%s), error: %v",
+				config.PRMergeBranch, err)
+		}
+
+		arg := strings.TrimSuffix(config.PRMergeBranch, "/merge")
+		if err := run(Git.Checkout(arg)); err != nil {
+			return fmt.Errorf("checkout failed (%s), error: %v", config.BranchDest, err)
+		}
+	} else if patch, err := getDiffFile(); err == nil {
+		if err := run(Git.Checkout(config.BranchDest)); err != nil {
+			return fmt.Errorf("checkout failed (%s), error: %v", config.BranchDest, err)
+		}
+		if err := run(Git.Apply(patch)); err != nil {
+			return fmt.Errorf("can't apply patch (%s), error: %v", patch, err)
+		}
+	} else {
+		return fmt.Errorf("there is no Pull Request branch and can't download diff file")
+	}
+	return nil
+}
+
+func manualMerge() error {
+	if err := runWithRetry(func() *command.Model {
+		if config.CloneDepth != "" {
+			return Git.Fetch("--depth=" + config.CloneDepth)
+		}
+		return Git.Fetch()
+	}); err != nil {
+		return fmt.Errorf("Fetch failed, error: %v", err)
+	}
+
+	if err := run(Git.Checkout(config.BranchDest)); err != nil {
+		return fmt.Errorf("checkout failed (%s), error: %v", config.BranchDest, err)
+	}
+
+	if isFork() {
+		if err := run(Git.RemoteAdd("upstream", config.PRRepositoryCloneURL)); err != nil {
+			return fmt.Errorf("couldn't add remote (%s), error: %v", config.PRRepositoryCloneURL, err)
+		}
+
+		if err := runWithRetry(func() *command.Model {
+			return Git.Fetch("upstream", config.Branch)
+		}); err != nil {
+			return fmt.Errorf("fetch Pull Request branch failed (%s), error: %v",
+				config.Branch, err)
+		}
+
+		if err := run(Git.Merge("upstream/" + config.Branch)); err != nil {
+			return fmt.Errorf("merge failed (upstream/%s), error: %v", config.Branch, err)
+		}
+	} else {
+		if err := run(Git.Merge(config.Commit)); err != nil {
+			return fmt.Errorf("merge failed (%s), error: %v", config.Commit, err)
+		}
+	}
+
+	return nil
+}
+
+func checkout(arg string) error {
+	if err := runWithRetry(func() *command.Model {
+		if config.CloneDepth != "" {
+			return Git.Fetch("--depth=" + config.CloneDepth)
+		}
+		return Git.Fetch()
+	}); err != nil {
+		return fmt.Errorf("Fetch failed, error: %v", err)
+	}
+
+	if err := run(Git.Checkout(arg)); err != nil {
+		if config.CloneDepth == "" {
+			return fmt.Errorf("checkout failed (%s), error: %v", checkoutArg, err)
```

Comment:
Where does `checkoutArg` come from? this function's input is `arg` not `checkoutArg`, and the first Checkout does use `arg` and not `checkoutArg`

Please avoid global variable as much as possible, other than maybe a single `config`!!